tests/test_telemetry.py

```diff
@@ -496,13 +522,43 @@ def test_app_product_change(self):
             raise Exception("app-product-change is not emitted when product change is enabled")
 
 
-@irrelevant(True, reason="The test must be adapted to handle correctly v1/v2 payloads")
-@released(python="1.7.0", dotnet="2.12.0", java="0.108.1", nodejs="3.2.0", ruby="1.4.0")
-@bug(context.uds_mode and context.library < "nodejs@3.7.0")
-@missing_feature(library="cpp")
-@missing_feature(library="php")
+@released(cpp="?", dotnet="2.35.0", golang="1.49.1", java="?", python="1.17.3", nodejs="?", php="0.90", ruby="1.11")
+class Test_TelemetryV2:
+    """Test telemetry v2 specific constraints"""
+
+    @missing_feature(library="golang", reason="Product started missing")
+    @missing_feature(library="dotnet", reason="Product started missing")
```

Comment:
This feature should be working for .NET 🤔

Reply:
Looks like this test is looking at `application.products`, when dotnet puts the data in `payload.products` . The ProductDisabled scenario does the opposite 🤷  which I missed.
I don't remember exactly why we have both... After checking we deserialize both value and merge them in the backend.

I really want to be done with this PR, so once this is merged, I'll open a second PR that looks at both locations for products for this test and the ProductDisabled scenario, and remove the annotation.
Is that fine?

Reply:
>Looks like this test is looking at `application.products`,

[That's the "v1" location](https://github.com/DataDog/instrumentation-telemetry-api-docs/blob/main/GeneratedDocumentation/ApiDocs/v1/SchemaDocumentation/Schemas/application.md#application) 🙂 

[In "v2" it _only_ should be on `payload.products`](https://github.com/DataDog/instrumentation-telemetry-api-docs/blob/main/GeneratedDocumentation/ApiDocs/v2/SchemaDocumentation/Schemas/app_started.md) 

> I really want to be done with this PR, so once this is merged, I'll open a second PR that looks at both locations for products for this test and the ProductDisabled scenario, and remove the annotation. Is that fine?

Sure, whatever works for you 😄